app/services/preserved_object_handler.rb

```diff
@@ -162,24 +153,37 @@ def create_db_objects(status, validated=false)
 
   def update_online_version
     results = []
-    pres_object = PreservedObject.find_by!(druid: druid)
-    pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint)
-    # FIXME: what if there is more than one associated pres_copy?
-    if incoming_version > pres_copy.version && pres_copy.version == pres_object.current_version
-      results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_copy.class.name)
-      update_preserved_copy(pres_copy, incoming_version, incoming_size)
-      results.concat(update_status(pres_copy, Status.ok))
-      results.concat(update_db_object(pres_copy))
-      results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_object.class.name)
-      update_preserved_object(pres_object, incoming_version)
-      results.concat(update_db_object(pres_object))
-    else
-      results << result_hash(UNEXPECTED_VERSION, 'PreservedCopy')
-      results.concat(version_comparison_results(pres_copy, :version))
-      results.concat(version_comparison_results(pres_object, :current_version))
-      # FIXME: TODO: should it update existence check timestamps/status?
+    upd_results = with_active_record_rescue do
+      pres_object = PreservedObject.find_by!(druid: druid)
+      pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint)
+      # FIXME: what if there is more than one associated pres_copy?
+      if incoming_version > pres_copy.version && pres_copy.version == pres_object.current_version
+        results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_copy.class.name)
+        update_preserved_copy(pres_copy, incoming_version, incoming_size)
+        results.concat(update_status(pres_copy, Status.ok))
+        results.concat(update_db_object(pres_copy))
+        results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_object.class.name)
+        update_preserved_object(pres_object, incoming_version)
+        results.concat(update_db_object(pres_object))
+      else
+        results << result_hash(UNEXPECTED_VERSION, 'PreservedCopy')
+        results.concat(version_comparison_results(pres_copy, :version))
+        results.concat(version_comparison_results(pres_object, :current_version))
+        # FIXME: TODO: should it update existence check timestamps/status?
+      end
     end
-    results
+    results.concat(upd_results)
+  end
+
+  def confirm_version_on_db_objects
```

Comment:
i am not very happy with this name, since it's so similar to `confirm_version_on_db_object`, which it calls.  i'm totally open to suggestions for better names.

Reply:
how about `confirm_version_in_db` or `confirm_version_in_catalog` ?